workers_launch_template.tf

```
@@ -1,157 +1,59 @@
-    var.worker_groups_launch_template[count.index],
-    "suspended_processes",
-    local.workers_group_defaults["suspended_processes"]
-  )
-  enabled_metrics = lookup(
-    var.worker_groups_launch_template[count.index],
-    "enabled_metrics",
-    local.workers_group_defaults["enabled_metrics"]
-  )
-  placement_group = lookup(
-    var.worker_groups_launch_template[count.index],
-    "placement_group",
-    local.workers_group_defaults["placement_group"],
-  )
-  termination_policies = lookup(
-    var.worker_groups_launch_template[count.index],
-    "termination_policies",
-    local.workers_group_defaults["termination_policies"]
-  )
-  max_instance_lifetime = lookup(
-    var.worker_groups_launch_template[count.index],
-    "max_instance_lifetime",
-    local.workers_group_defaults["max_instance_lifetime"],
-  )
-  default_cooldown = lookup(
-    var.worker_groups_launch_template[count.index],
-    "default_cooldown",
-    local.workers_group_defaults["default_cooldown"]
-  )
-  health_check_type = lookup(
-    var.worker_groups_launch_template[count.index],
-    "health_check_type",
-    local.workers_group_defaults["health_check_type"]
-  )
-  health_check_grace_period = lookup(
-    var.worker_groups_launch_template[count.index],
-    "health_check_grace_period",
-    local.workers_group_defaults["health_check_grace_period"]
-  )
+  desired_capacity          = each.value.asg_desired_capacity
+  max_size                  = each.value.asg_max_size
+  min_size                  = each.value.asg_min_size
+  force_delete              = each.value.asg_force_delete
+  target_group_arns         = each.value.target_group_arns
+  load_balancers            = each.value.load_balancers
+  service_linked_role_arn   = each.value.service_linked_role_arn
+  vpc_zone_identifier       = each.value.subnets
+  protect_from_scale_in     = each.value.protect_from_scale_in
+  suspended_processes       = each.value.suspended_processes
+  enabled_metrics           = each.value.enabled_metrics
+  placement_group           = each.value.placement_group
+  termination_policies      = each.value.termination_policies
+  max_instance_lifetime     = each.value.max_instance_lifetime
+  default_cooldown          = each.value.default_cooldown
+  health_check_type         = each.value.health_check_type
+  health_check_grace_period = each.value.health_check_grace_period
 
   dynamic "mixed_instances_policy" {
     iterator = item
-    for_each = (lookup(var.worker_groups_launch_template[count.index], "override_instance_types", null) != null) || (lookup(var.worker_groups_launch_template[count.index], "on_demand_allocation_strategy", local.workers_group_defaults["on_demand_allocation_strategy"]) != null) ? [var.worker_groups_launch_template[count.index]] : []
+
+    for_each = (lookup(var.worker_groups_launch_template[each.key], "override_instance_types", null) != null) || (each.value.on_demand_allocation_strategy != null) ? list(each.value) : []
```

I think this (and many other similar lines) need to use `each.value` rather than `var.worker_groups_launch_template[each.key]`

```suggestion
    for_each = (lookup(each.value, "override_instance_types", null) != null) || (each.value.on_demand_allocation_strategy != null) ? list(each.value) : []
```